core/inc/knoting/audio_listener.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <knoting/components.h>
+#include <knoting/types.h>
+#include <fmod.hpp>
+
+namespace knot::components {
+
+class AudioListener {
+   public:
+    AudioListener() = default;
+    ~AudioListener() = default;
+    AudioListener(const AudioListener& other) = default;
+    //    //=For ECS========
+    //    void on_awake() {}
+    //    void on_destroy() {}
+    //    //================
+
+    FMOD_VECTOR* get_position();
+    quat get_rotation();
+    FMOD_VECTOR* get_forward();
+    FMOD_VECTOR* get_up();
```

Comment:
Try to only expose data types that are also used in other systems, so instead of a FMOD_VECTOR* return a `vec3` and do any convertion needed in the subsystem side